ext/otel_observer.c

```diff
@@ -917,7 +930,9 @@ static void copy_observer(otel_observer *source, otel_observer *destination) {
 
 static bool find_observers(HashTable *ht, zend_string *n, zend_llist *pre_hooks,
                            zend_llist *post_hooks) {
-    otel_observer *observer = zend_hash_find_ptr_lc(ht, n);
+    zend_string *normalized = normalize_fq_func_name(n);
```

Comment:
Should instead strip the leading `\` in `add_function_observer` to avoid the allocations during lookup.